content/push/device.textile

```diff
@@ -15,109 +16,114 @@ redirect_from:
+
+dependencies {
+  implementation 'io.ably:ably-android:1.2.0'
+}
+
+repositories {
+  mavenCentral()
+}
 ```
 
-h3. Integrate Ably into your app
+```[swift]
+// The Ably Client Library SDK is available as a CocoaPod.
+// Add the SDK to your Podfile:
 
-The following steps guide you through the "Ably SDK":https://ably.com/docs/sdks integration process:
+pod 'Ably'
 
-```[android]
-// Define the custom action for push notifications
-public class AblyPushNotificationService extends FirebaseMessagingService {
+// Install the SDK with:
 
-    // This method is overridden to handle any messages received while the app is in the foreground or background.
-    @Override
-    public void onMessageReceived(@NonNull RemoteMessage message) {
-        // Process the incoming message here
-    }
+pod install
 
-    //This method is overridden to handle token refresh, ensuring that Ably is always updated with the latest token.
-    @Override
-    public void onNewToken(@NonNull String token) {
-        super.onNewToken(s);
-        // Update Ably's server with the new token
-        ActivationContext.getActivationContext(this).onNewRegistrationToken(RegistrationToken.Type.FCM, s);
-    }
-}
+// Import the SDK into your files with:
+
+import Ably
 ```
 
-```[swift]
-// Extend your UIApplicationDelegate to handle remote notification registration lifecycle events.
-func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
-    // This method is called when APNs has successfully registered the device for push notifications
-    ARTPush.didRegisterForRemoteNotifications(withDeviceToken: deviceToken, realtime: self.getAblyRealtime())
-}
+```[objc]
+// The Ably Client Library SDK is available as a CocoaPod.
+// Add the SDK to your Podfile:
 
-func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
-    // This method is called when APNs failed to register the device for push notifications
-    ARTPush.didFailToRegisterForRemoteNotificationsWithError(error, realtime: self.getAblyRealtime())
-}
+pod 'Ably'
 
-// Helper function to configure and return an instance of ARTRealtime
-func getAblyRealtime() -> ARTRealtime {
-    var options = ARTClientOptions()
-    return ARTRealtime(options: options)
-}
+// Install the SDK with:
+
+pod install
 
+// Import the SDK into your files with:
+
+#import "Ably.h"
 ```
 
-```[objc]
-// Extend your UIApplicationDelegate to handle remote notification registration lifecycle events.
-- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
-// This method is called when APNs has successfully registered the device for push notifications
-    [ARTPush didRegisterForRemoteNotificationsWithDeviceToken:deviceToken realtime:[self getAblyRealtime]];
-}
+```[csharp]
```

Comment:
.NET has [specific packages](https://github.com/ably/ably-dotnet/blob/main/PushNotifications.md#setting-up-the-dotnetpush-app) for Android and iOS so this needs to be updated.